gems/sorbet-runtime/test/types/method_patches.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+require_relative '../test_helper'
+
+module Opus::Types::Test
+  class MethodPatchesTest < Critic::Unit::UnitTest
+    before do
+      @mod = Module.new do
+        extend T::Sig
+        # Make it public for testing only
+        public_class_method :sig
+      end
+    end
+
+    describe "arity" do
+      it "succeeds on method without signature" do
+        def @mod.foo(bar)
+          :foo
+        end
+
+        method = @mod.method(:foo)
+        assert_equal(1, method.arity)
+      end
+
+      it "succeeds on method with signature" do
+        @mod.sig {params(bar: T.untyped).returns(T.untyped)}
+        def @mod.foo(bar)
+          :foo
+        end
+
+        method = @mod.method(:foo)
+        assert_equal(1, method.arity)
+      end
```

Comment:
Just a suggestion, feel free to ignore: what do you think about having each test declare both a sigged and unsigged method, and assert that the arity of one method is equal to the other? I think the property we care about is not so much that this method's arity is `1`, but that it's the same regardless of a sig.

Reply:
Thanks, that was a good suggestion. I did basically that, except for the `source_location` method. I guess I could have done that for that method too, but somehow it felt cleaner to check that the file was matching the current file.

But now that I am writing this, I am having second thoughts since the implementation is unnecessarily complicating the test code.

Reply:
Ok, done!